sos/report/plugins/vault.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (C) 2023 Canonical Ltd., Arif Ali <arif.ali@canonical.com>
+#
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, UbuntuPlugin
+
+
+class Vault(Plugin, UbuntuPlugin):
+    """The Vault plugin collects the current status of the vault
+    snap on a Ubuntu machine.
+
+    It will collect logs from journal, vault status and configuration
+    """
+
+    short_desc = 'Manage access to secrets and protect sensitive data'
+
+    plugin_name = 'vault'
+    profiles = ('sysmgmt', 'security')
+    package = ('vault',)
+
+    def setup(self):
+
+        vault_cfg = "/var/snap/vault/common/vault.hcl"
+
+        self.add_service_status('vault')
+
+        self.add_journal('vault')
```

Comment:
If you use the `services = ('vault',)` class attr these collections become implicit.